utility/types/tx_fifo_mempool_test.go

```diff
@@ -0,0 +1,298 @@
+package types
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pokt-network/pocket/shared/codec"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTxFIFOMempool(t *testing.T) {
+	type args struct {
+		maxTxBytes uint64
+		maxTxs     uint32
+		initialTxs *[][]byte
```

Comment:
1. Remove pointer?
1.1 Why are we doing a pointer to a slice? When slices are passed around, it's already just a reference to the first element.
1.2 If we do this, we should do it for `actions` too

2. If we don't do this, why is this a pointer but `wantItems` is not?

The diagram here helped me think through it: https://medium.com/swlh/golang-tips-why-pointers-to-slices-are-useful-and-how-ignoring-them-can-lead-to-tricky-bugs-cac90f72e77b

![Screenshot 2023-02-02 at 5 18 58 PM](https://user-images.githubusercontent.com/1892194/216488091-cd512869-2448-4f47-a41b-ba6d46f7bd31.png)

Reply:
I normally use pointers in tests like these when I feel like I might not want to configure some cases with `InitialTxs`

Basically to provide that flexibility to the tester.

![image](https://user-images.githubusercontent.com/29378614/216635204-e30f6ce4-0a34-465a-843e-5eb030e1043a.png)

if nil we just skip that.

Reply:
I see the use case and raise you this mini demo:

![Screenshot 2023-02-03 at 10 52 16 AM](https://user-images.githubusercontent.com/1892194/216683831-1da9e75d-157a-48bb-b234-8c128bbaa958.png)

If we just don't set a slice, it's nil by default.

I would request one of following:
1. Document (in the struct next to the field) why it it's a pointer
2. Not use a pointer at all

Source code for reference:
```go
package main

import "fmt"

func main() {
	type args struct {
		ptrSlice *[]byte
		slice    []byte
	}

	a := &args{
		ptrSlice: &[]byte{},
		slice:    []byte{},
	}

	b := &args{}

	fmt.Println("a.ptrSlice:", a.ptrSlice == nil)
	fmt.Println("a.slice   :", a.slice == nil)

	fmt.Println("b.ptrSlice:", b.ptrSlice == nil)
	fmt.Println("b.slice   :", b.slice == nil)
}
```

Reply:
💯 but I meant it to be used like this:

![image](https://user-images.githubusercontent.com/29378614/216694279-8a79a1f8-043b-49fa-91c7-b2361e157323.png)